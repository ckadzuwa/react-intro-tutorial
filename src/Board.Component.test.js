import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import Board from './Board.Component'
import Square from './Square.Component'

configure({ adapter: new Adapter() });

test('it should render without crashing', () => {
    const board = shallow(<Board/>);
    expect(board).toHaveLength(1);
});

test('it should match the snapshot', () => {
    const board = shallow(<Board/>);
    expect(board).toMatchSnapshot();
});

test('it should 3x3 grid giving a total of nine squares', () => {
    const board = shallow(<Board/>);
    expect(board.find('.board-row')).toHaveLength(3);
    expect(board.find(Square)).toHaveLength(9);
});

test('it should update internal state when a square is clicked', () => {
    const board = mount(<Board/>);
    const firstSquare = board.find(Square).first();
    expect(board.state('squares')).toEqual(Array(9).fill(null));
   
    firstSquare.find('button').simulate('click');

    expect(board.state('squares')).toEqual(['X', ...Array(8).fill(null)]);
});

test('it should update the sqaure clicked with an X', () => {
    const board = mount(<Board/>);
    const firstSquare = board.find(Square).first();
    expect(firstSquare.html()).toEqual('<button class=\"square\"></button>');

    firstSquare.find('button').simulate('click');

    expect(firstSquare.html()).toEqual('<button class=\"square\">X</button>');
});