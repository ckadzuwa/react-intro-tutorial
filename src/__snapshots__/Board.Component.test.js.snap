// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Board />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="status"
        >
          Next player: X
        </div>,
        <div
          className="board-row"
        >
          <Square
            onClick={[Function]}
            value={null}
          />
          <Square
            onClick={[Function]}
            value={null}
          />
          <Square
            onClick={[Function]}
            value={null}
          />
        </div>,
        <div
          className="board-row"
        >
          <Square
            onClick={[Function]}
            value={null}
          />
          <Square
            onClick={[Function]}
            value={null}
          />
          <Square
            onClick={[Function]}
            value={null}
          />
        </div>,
        <div
          className="board-row"
        >
          <Square
            onClick={[Function]}
            value={null}
          />
          <Square
            onClick={[Function]}
            value={null}
          />
          <Square
            onClick={[Function]}
            value={null}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Next player: X",
          "className": "status",
        },
        "ref": null,
        "rendered": "Next player: X",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Square
              onClick={[Function]}
              value={null}
            />,
            <Square
              onClick={[Function]}
              value={null}
            />,
            <Square
              onClick={[Function]}
              value={null}
            />,
          ],
          "className": "board-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Square
              onClick={[Function]}
              value={null}
            />,
            <Square
              onClick={[Function]}
              value={null}
            />,
            <Square
              onClick={[Function]}
              value={null}
            />,
          ],
          "className": "board-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Square
              onClick={[Function]}
              value={null}
            />,
            <Square
              onClick={[Function]}
              value={null}
            />,
            <Square
              onClick={[Function]}
              value={null}
            />,
          ],
          "className": "board-row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onClick": [Function],
              "value": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="status"
          >
            Next player: X
          </div>,
          <div
            className="board-row"
          >
            <Square
              onClick={[Function]}
              value={null}
            />
            <Square
              onClick={[Function]}
              value={null}
            />
            <Square
              onClick={[Function]}
              value={null}
            />
          </div>,
          <div
            className="board-row"
          >
            <Square
              onClick={[Function]}
              value={null}
            />
            <Square
              onClick={[Function]}
              value={null}
            />
            <Square
              onClick={[Function]}
              value={null}
            />
          </div>,
          <div
            className="board-row"
          >
            <Square
              onClick={[Function]}
              value={null}
            />
            <Square
              onClick={[Function]}
              value={null}
            />
            <Square
              onClick={[Function]}
              value={null}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Next player: X",
            "className": "status",
          },
          "ref": null,
          "rendered": "Next player: X",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Square
                onClick={[Function]}
                value={null}
              />,
              <Square
                onClick={[Function]}
                value={null}
              />,
              <Square
                onClick={[Function]}
                value={null}
              />,
            ],
            "className": "board-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Square
                onClick={[Function]}
                value={null}
              />,
              <Square
                onClick={[Function]}
                value={null}
              />,
              <Square
                onClick={[Function]}
                value={null}
              />,
            ],
            "className": "board-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Square
                onClick={[Function]}
                value={null}
              />,
              <Square
                onClick={[Function]}
                value={null}
              />,
              <Square
                onClick={[Function]}
                value={null}
              />,
            ],
            "className": "board-row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onClick": [Function],
                "value": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
